### PR TITLE
Refactor `functions.php`

### PR DESCRIPTION
**Pull request by everly-gif**
## What does this PR do?

Refactors `functions.php`

## Test Plan

Manual

- checked if the IDE is still able to pickup params + builds locally

## Related PRs and Issues

NIL

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

yes

### CODE REVIEW

**Comment by everly-gif** [comment]
Need a **_careful review_** on the optional parameteres.

**Review by eldadfux — APPROVED** [review]

File: app/controllers/api/functions.php [hunk 1]
```diff
@@ -191,11 +187,7 @@
     ->param('range', '30d', new WhiteList(['24h', '7d', '30d', '90d']), 'Date range.', true)
     ->inject('response')
     ->inject('dbForProject')
-    ->action(function ($functionId, $range, $response, $dbForProject) {
-        /** @var Appwrite\Utopia\Response $response */
-        /** @var Utopia\Database\Document $project */
-        /** @var Utopia\Database\Database $dbForProject */
-        /** @var Utopia\Registry\Registry $register */
+    ->action(function (string $functionId, ?string $range, Response $response, Database $dbForProject) {
```
Comment (TorstenDittmann) [inline app/controllers/api/functions.php:190]: Not sure if `?string` makes sense here?

Since it will never be `null` and is always set to `30d`.

Comment (eldadfux) [reply app/controllers/api/functions.php:190]: I think as long as it's set as optional we should be strict and keep it as is.

Comment (TorstenDittmann) [reply app/controllers/api/functions.php:190]: But it's never optional here at this point, it will always be what is set as default in the param.

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**